src/main/java/seedu/lifeasier/model/tasks/Deadline.java

```diff
@@ -70,6 +72,19 @@ public void moveAndUpdateRecurrences(LocalDate day) {
             decrementRecurrences(1);
             by = by.plusDays(7);
         }
+    }
+
+    public boolean isDuplicate(String description, LocalDateTime by, int recurrences) {
+
+        LocalTime existingBy = this.by.toLocalTime();
+        DayOfWeek existingDay = this.by.getDayOfWeek();
+
+        LocalTime newBy = by.toLocalTime();
+        DayOfWeek newDay = by.getDayOfWeek();
 
+        return (this.description.equals(description)
+                && existingBy == newBy
+                && existingDay == newDay
+                && ((this.recurrences > 0 && recurrences > 0) || this.by.equals(by)));
```

Comment:
Maybe we can focus on readability more for this part instead of valuing performance, so we can change this to a boolean value and return it instead. 
```suggestion
       boolean isDuplicate = (this.description.equals(description)
                && existingBy == newBy
                && existingDay == newDay
                && ((this.recurrences > 0 && recurrences > 0) || this.by.equals(by)));
      return isDuplicate;
```

Reply:
Doing that causes IntelliJ to give me a warning that the boolean is unnecessary. I'll add a Javadoc comment instead to better explain what is happening. I hope that's sufficient!